sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -312,6 +313,10 @@ highlightFunction
     : HIGHLIGHT LR_BRACKET relevanceField (COMMA highlightArg)* RR_BRACKET
     ;
 
+positionFunction
+    : POSITION LR_BRACKET functionArg IN functionArg RR_BRACKET
+    ;
```

Comment:
I think we just need to add position function name to `textFunctionName` rule? Same for changes in AST builder, analyzer and expression. I assume nothing special in position() and thus it should be handled by general visitFunction() in AST builder and analyzer. And also no need to add `PositionExpression` for it. Please check any other string function to confirm.

Reply:
This function has the following syntax: `POSITION(substr IN str)` that contains `IN` , unlike other string functions that have comma separated list of arguments. I was not able to find any other string function with similar syntax, and `IN` seems not to be handled in case position is added to `textFunctionName`.

Reply:
I see. Sorry I missed that. Then is it possible to eliminate this difference in AST layer? As I understand this is more like syntax sugar. After AST, it can be converted to a regular string function call with arguments [substr, str] ?

Reply:
How currently this works in SQL is as follows: 
```opensearchsql> select position('H', 'jjjjH');                                                                                                         
{'reason': 'Invalid SQL query', 'details': 'syntax error, expect IN, actual COMMA H', 'type': 'ParserException'}```
However, the syntax with `IN` works as expected.
Currently, I don't see a better way to implement the AST layer. Please advise how would you suggest to alter the implementation. Thank you!

Reply:
Thank you @dai-chen for the quick response! Addressed in 3786f04ae04c91df1d5befa728ea9f7b2eea6a48 commit.